### ISSUE TITLE
[bug] user_home_short=None doesn't work

### ISSUE THREAD

**Issue by sourcedelica** [conan-io/conan#7287]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: Visual Studio
  * Conan version: 1.26.1
  * Python version: 3.7

### Steps to reproduce (Include if Applicable)
Add `user_home_short = None` to `conan.conf`.  Conan still uses short paths for the recipes that use them.  Setting `CONAN_USER_HOME_SHORT=None` does work.

### Logs (Executed commands with output) (Include/Attach if Applicable)
```
$ conan config get
[log]
run_to_output = True        # environment CONAN_LOG_RUN_TO_OUTPUT
run_to_file = False         # environment CONAN_LOG_RUN_TO_FILE
level = 50                  # environment CONAN_LOGGING_LEVEL
print_run_commands = False  # environment CONAN_PRINT_RUN_COMMANDS

[general]
default_profile = default
compression_level = 9                 # environment CONAN_COMPRESSION_LEVEL
sysrequires_sudo = True               # environment CONAN_SYSREQUIRES_SUDO
request_timeout = 60                  # environment CONAN_REQUEST_TIMEOUT (seconds)
user_home_short = None
revisions_enabled = True

[storage]
path = ~/.conan/data

[hooks]
attribute_checker
```

```
$ export -n CONAN_USER_HOME_SHORT

$ conan info .. --paths --package-filter="protobuf*"
    ID: 6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7
    BuildID: None
    export_folder: C:\Users\epederson\.conan\data\protobuf\3.6.1\bincrafters\stable\export
    source_folder: C:\.conan\c7ab61\1
    build_folder: C:\.conan\220f65\1
    package_folder: C:\.conan\93c189\1
    Remote: conan-idb-third-party=http://usdlvart01:8081/artifactory/api/conan/conan-idb-third-party
    URL: https://github.com/bincrafters/conan-protobuf
    Homepage: https://github.com/protocolbuffers/protobuf
    License: BSD-3-Clause
    Author: Bincrafters <bincrafters@gmail.com>
    Description: Protocol Buffers - Google's data interchange format
    Topics: conan, protobuf, protocol-buffers, protocol-compiler, serialization, rpc, protocol-compiler
    Recipe: Cache
    Revision: 0
    Binary: Cache
    Binary remote: conan-center
    Creation date: 2019-08-13 12:30:09
    Required by:
        conanfile.py
```

**Comment by jgsogo** [comment]
Thanks for reporting.

Indeed it looks like that `user_home_short` is not taken into account at all  😮  Interesting bits here in the property `short_paths_home` (https://github.com/conan-io/conan/blob/develop/conans/client/conf/__init__.py#L492), but later it is not used in when actually using the [short home for Windows](https://github.com/conan-io/conan/blob/b9789e3ab85609358af41f63d5599750b6dc14c5/conans/util/windows.py#L62).

**Comment by memsharded** [comment]
Fixed in https://github.com/conan-io/conan/pull/9118, will be released in 1.38
